### PR TITLE
Added new config setting: DelayChannelAutocomplete

### PR DESCRIPTION
**Pull request by cwarnermm**
Documentation for: 
- https://github.com/mattermost/mattermost/pull/22952
- https://github.com/mattermost/mattermost-mobile/pull/7297

### CODE REVIEW

File: source/configure/experimental-configuration-settings.rst [hunk 1]
```diff
@@ -660,6 +660,19 @@ See the `Important Upgrade Notes <https://docs.mattermost.com/upgrade/important-
 | This feature's ``config.json`` setting is ``"PatchPluginsReactDOM": false`` with options ``true`` and ``false``. |
 +------------------------------------------------------------------------------------------------------------------+
 
+Delay channel autocomplete
```
Comment (cwarnermm) [inline source/configure/experimental-configuration-settings.rst:663]: @hmhealey - Is this new config setting available to both self-hosted and Cloud deployments? And is it available in TE or does it require Professional or Enterprise?

Comment (hmhealey) [reply source/configure/experimental-configuration-settings.rst:663]: It's available everywhere. There's no license or location requirements on it

**Review by hmhealey — APPROVED** [review]
I actually already had a PR submitted for this one that's been blocked on the new release. Should I close it in favour of this one? https://github.com/mattermost/docs/pull/6346

File: source/configure/experimental-configuration-settings.rst [hunk 2]
```diff
@@ -660,6 +660,19 @@ See the `Important Upgrade Notes <https://docs.mattermost.com/upgrade/important-
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This setting makes the channel autocomplete, such as ``~town-square`` less obtrusive for people who uses tildes ``~`` as punctuation. 
+
+**True**: The autocomplete for channel links only triggers after typing a tilde symbol followed by two or more characters.
```
Comment (cwarnermm) [inline source/configure/experimental-configuration-settings.rst:668]:
```suggestion
**True**: The autocomplete appears after the user types a tilde followed by two or more characters. For example, typing ``~to`` will show the autocomplete, but typing ``~`` will not. 
```

File: source/configure/experimental-configuration-settings.rst [hunk 3]
```diff
@@ -660,6 +660,19 @@ See the `Important Upgrade Notes <https://docs.mattermost.com/upgrade/important-
...
 +------------------------------------------------------------------------------------------------------------------+
 
+Delay channel autocomplete
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This setting makes the channel autocomplete, such as ``~town-square`` less obtrusive for people who uses tildes ``~`` as punctuation. 
+
+**True**: The autocomplete for channel links only triggers after typing a tilde symbol followed by two or more characters. 
+
+**False**: (Default) The autocomplete for channel links appears as soon as the user types a tilde symbol.
```
Comment (cwarnermm) [inline source/configure/experimental-configuration-settings.rst:670]:
```suggestion
**False**: **(Default)** The autocomplete appears immediately after the user types a tilde. For example, typing ``~`` will show the autocomplete.
```

**Pull request merged**